var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -348,18 +361,45 @@ def cmake_args(self):
     def pre_install(self):
         with working_dir(self.build_directory):
             # When building shared libraries these need to be installed first
-            make('install-LLVMTableGen')
+            ninja('install-LLVMTableGen')
             if self.spec.version >= Version('4.0.0'):
                 # LLVMDemangle target was added in 4.0.0
-                make('install-LLVMDemangle')
-            make('install-LLVMSupport')
+                ninja('install-LLVMDemangle')
+            ninja('install-LLVMSupport')
 
     @run_after('install')
     def post_install(self):
+        spec = self.spec
+
+        # unnecessary if we get bootstrap builds in here
+        if '+cuda' in self.spec:
+            ompdir = 'build-bootstrapped-omp'
+            # rebuild libomptarget to get bytecode runtime library files
+            with working_dir(ompdir, create=True):
+                cmake_args = [
+                    self.stage.source_path + '/openmp',
+                      '-DCMAKE_C_COMPILER:PATH={0}'.format(spec.prefix.bin + '/clang'),
+                      '-DCMAKE_CXX_COMPILER:PATH={0}'.format(spec.prefix.bin + '/clang++'),
+                      '-DCMAKE_INSTALL_PREFIX:PATH={0}'.format(spec.prefix),
+                              ]
+                cmake_args.append('-DCMAKE_BUILD_TYPE:String={0}'.format(spec.variants['build_type'].value))
+                # work around bad libelf detection in libomptarget
+                cmake_args.append('-DCMAKE_CXX_FLAGS:String=-I{0}'.format(spec['libelf'].prefix.include))
+                cmake_args.append('-DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES:LIST={0}'.format(','.join(spec.variants['nvptx_offload_ccs'].value)))
+                cmake_args.append('-DCUDA_TOOLKIT_ROOT_DIR:PATH=' + spec['cuda'].prefix)
+                cmake_args.append('-DCUDA_NVCC_EXECUTABLE:FILEPATH=' + spec['cuda'].prefix.bin + '/nvcc')
```

Comment:
You can take a short-cut here and recycle previous compiler flags in the post-install method. Here is what I did for `llvm-flang`
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/llvm-flang/package.py#L221-L239

Reply:
Good point, reworked.